File: cs/sdk/src/Proton.Drive.Sdk/Nodes/Revision.cs
using Proton.Sdk;

namespace Proton.Drive.Sdk.Nodes;

public sealed record Revision
{
    public required RevisionUid Uid { get; init; }
    public required DateTime CreationTime { get; init; }
    public required long SizeOnCloudStorage { get; init; }
    public required long? ClaimedSize { get; init; }
    public required DateTime? ClaimedModificationTime { get; init; }
    public required IReadOnlyList<ReadOnlyMemory<byte>> Thumbnails { get; init; }
    public required Result<Author, SignatureVerificationError>? ContentAuthor { get; init; }
}

File: kt/sdk/src/main/kotlin/me/proton/drive/sdk/entity/Revision.kt
package me.proton.drive.sdk.entity

data class FileRevision(
    val uid: String,
    val creationTime: Long,
    val sizeOnCloudStorage: Long,
    val claimedSize: Long?,
    val claimedDigests: FileContentDigests,
    val claimedModificationTime: Long?,
    val thumbnails: List<ThumbnailHeader>,
    val additionalClaimedMetadata: List<AdditionalMetadataProperty>?,
    val contentAuthor: Result<Author>?,
)
